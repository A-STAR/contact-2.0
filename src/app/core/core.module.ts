import { ErrorHandler, NgModule, Optional, SkipSelf } from '@angular/core';
import { DatePipe } from '@angular/common';

import { JwtHelper } from 'angular2-jwt';
import { TranslateService } from '@ngx-translate/core';
import { StoreModule } from '@ngrx/store';
import { EffectsModule } from '@ngrx/effects';
import { StoreDevtoolsModule } from '@ngrx/store-devtools';

import { environment } from '../../environments/environment';

import { AuthEffects } from './auth/auth.effects';
import { AuthHttpService } from './auth/auth-http.service';
import { AuthService } from './auth/auth.service';
import { ContentTabService } from '../shared/components/content-tabstrip/tab/content-tab.service';
import { DataService } from './data/data.service';
import { DictionariesEffects } from './dictionaries/dictionaries.effects';
import { DictionariesService } from './dictionaries/dictionaries.service';
import { EntityTranslationsService } from './entity/translations/entity-translations.service';
import { ErrorHandlerService } from './error/error-handler.service';
import { LookupEffects } from './lookup/lookup.effects';
import { LookupService } from './lookup/lookup.service';
import { MenuService } from './menu/menu.service';
import { MetadataService } from './metadata/metadata.service';
import { NotificationsEffects } from './notifications/notifications.effects';
import { NotificationsService } from './notifications/notifications.service';
import { SettingsService } from './settings/settings.service';
import { ThemesService } from './themes/themes.service';
import { throwIfAlreadyLoaded } from './module-import-guard';
import { MetadataEffects } from './metadata/metadata.effects';
import { UserConstantsEffects } from './user/constants/user-constants.effects';
import { UserConstantsService } from './user/constants/user-constants.service';
import { UserDictionariesEffects } from './user/dictionaries/user-dictionaries.effects';
import { UserDictionariesService } from './user/dictionaries/user-dictionaries.service';
import { UserLanguagesEffects } from './user/languages/user-languages.effects';
import { UserLanguagesService } from './user/languages/user-languages.service';
import { UserPermissionsEffects } from './user/permissions/user-permissions.effects';
import { UserPermissionsService } from './user/permissions/user-permissions.service';
import { ValueConverterService } from './converter/value/value-converter.service';

import { rootReducer } from './state/root.reducer';

@NgModule({
  imports: [
    StoreModule.provideStore(rootReducer),
    EffectsModule.run(AuthEffects),
    EffectsModule.run(DictionariesEffects),
    EffectsModule.run(LookupEffects),
    EffectsModule.run(NotificationsEffects),
    EffectsModule.run(UserConstantsEffects),
    EffectsModule.run(UserDictionariesEffects),
    EffectsModule.run(UserLanguagesEffects),
    EffectsModule.run(UserPermissionsEffects),
    EffectsModule.run(MetadataEffects),
    environment.production
      ? []
      : StoreDevtoolsModule.instrumentOnlyWithExtension({
          maxAge: 1024
        })
  ],
  providers: [
    AuthHttpService,
    AuthService,
    ContentTabService,
    EntityTranslationsService,
    DictionariesService,
    DatePipe,
    DataService,
    JwtHelper,
    LookupService,
    MenuService,
    MetadataService,
    NotificationsService,
    SettingsService,
    ThemesService,
    TranslateService,
    UserConstantsService,
    UserDictionariesService,
    UserLanguagesService,
    UserPermissionsService,
    ValueConverterService,
    {
      provide: ErrorHandler,
      useClass: ErrorHandlerService
    }
  ],
})
export class CoreModule {
  constructor(@Optional() @SkipSelf() parentModule: CoreModule) {
    throwIfAlreadyLoaded(parentModule, 'CoreModule');
  }
}
