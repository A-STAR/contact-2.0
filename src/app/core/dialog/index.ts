export abstract class DialogFunctions {
  /**
   * NOTE: this prop cannot come from the prototype and has to be implemented in the children
   *
   * @abstract
   * @type {string}
   * @memberof DialogFunctions
   */
  abstract dialog: string;

  isDialog(dialog: string): boolean {
    return this.dialog === dialog;
  }

  setDialog(dialog: string = null): void {
    this.dialog = dialog;
  }

  closeDialog(): void {
    this.setDialog();
  }
  /**
   * Same as `closeDialog` just to maintain the `on` prefix used in EventEmitters
   *
   * @memberof DialogFunctions
   */
  onCloseDialog(): void {
    this.setDialog();
  }

  onResult(result: any): void {

  }
}
