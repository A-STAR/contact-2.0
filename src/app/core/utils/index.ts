/**
 * General purpose utility library
 * Description: exports small utility functions to be used across different components
 */

import { IOption, INamedValue } from '../converter/value-converter.interface';

export const propOr = (prop: string, orValue: any) => obj => Object.hasOwnProperty.call(obj, prop) ? obj[prop] : orValue;

export const makeKey = (prefix: string) => (fieldName: string) => `${prefix}.${fieldName}`;

export const toLabeledValues = item => ({ label: item.name, value: item.code });

type IValueToOption<T> = (value: T) => IOption;

export const toOption = <T extends Object>(valueKey: keyof T, labelKey: keyof T): IValueToOption<T> => {
  return value => ({
    label: value[labelKey],
    value: value[valueKey]
  });
};

export const valuesToOptions = (values: Array<INamedValue>): Array<IOption> => {
  return (values || []).map(toOption('id', 'name'));
};

export const toFullName = (person: { lastName: string, firstName: string, middleName: string }) => {
  return [ person.lastName, person.firstName, person.middleName ]
    .filter(Boolean).join(' ');
};

export const timeToHourMinSec = (str: string): { hour: number; minute: number; second: number; } => {
  // const validate = obj => {};
  const [ hour, minute, second ] = str.split(':').map(Number);
  return { hour, minute, second };
};

export const arrayToObject = (key: string) => (arr: Array<any>) => {
  return arr.reduce((acc, item) => {
    acc[item[key]] = item;
    return acc;
  }, {});
};

export const checkboxRenderer = (key: string) => ({ [key]: truthy }) =>
  `<i class="fa fa${truthy ? '-check' : ''}-square-o" aria-hidden="true"></i>`;

export const yesNoRenderer = (key: string) => ({ [key]: truthy }) => truthy ? 'default.yesNo.Yes' : 'default.yesNo.No';

const reverseString = (str: string) => str.split('').reverse().join('');

export const phoneRenderer = (key: string) => ({ [key]: phone }) => (phone || '')
    .trim()
    .split('')
    .reverse()
    .join('')
    .replace(/^(\d{4})?(\d{1,3})?(\d{1,3})?(\d{1,3})?/, (str, p1, p2, p3, p4) => {
      const [t1, t2, t3, t4] = [p1, p2, p3, p4]
        .map(p => p || '')
        .map(reverseString)
        .reverse();
      return `${t1 ? '+' + t1 + ' ' : ''}${t2 ? '(' + t2 + ') ' : ''}${t3}-${t4}`;
  });

export const numberRenderer = (key: string) => ({ [key]: num }) => {
  const parts = (String(num == null ? 0 : num) || '').split('.');
  parts[0] = parts[0].replace(/\B(?=(\d{3})+(?!\d))/g, ',');
  if (!parts[1]) {
    parts.push('00');
  }
  return parts.join('.');
};

export const flatten = <T extends { children?: T[] }>(items: T[], key: keyof T = null) => {
  return items.reduce((acc, item) => [
    ...acc,
    key ? item[key] : item,
    ...(item.children ? flatten(item.children, key) : []),
  ], []);
};

export const isEmpty = (array: any[]): boolean => !array || array.length === 0;

export const parseStringValueAttrs = (str: string) => (str || '')
  .split(/,\s?/g)
  .filter(Boolean)
  .map(s => parseInt(s, 10))
  .filter(n => n > 0 && n <= 10 )
  .map(attr => `stringValue${attr}`);

export const renderers = {
  checkboxRenderer,
  phoneRenderer,
  yesNoRenderer,
  numberRenderer,
};

export const round = (value: number, precision: number) => {
  const k = Math.pow(10, precision);
  return Math.round(k * value) / k;
};
