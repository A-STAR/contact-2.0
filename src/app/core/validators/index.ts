import { hasDigits } from './hasDigits';
import { hasLowerCaseChars } from './hasLowerCaseChars';
import { hasUpperCaseChars } from './hasUpperCaseChars';
import { isDate } from './isDate';
import { max, maxStrict } from './max';
import { maxDate } from './maxDate';
import { maxFileSize } from './maxFileSize';
import { min, minStrict } from './min';
import { minDate, minDateThreeDaysAgo } from './minDate';
import { multilanguageRequired } from './multilanguageRequired';
import { oneOfGroupRequired } from './oneOfGroupRequired';
import { password } from './password';

export {
  hasDigits,
  hasLowerCaseChars,
  hasUpperCaseChars,
  isDate,
  max,
  maxDate,
  maxFileSize,
  maxStrict,
  min,
  minDate,
  minDateThreeDaysAgo,
  minStrict,
  multilanguageRequired,
  oneOfGroupRequired,
  password,
};
