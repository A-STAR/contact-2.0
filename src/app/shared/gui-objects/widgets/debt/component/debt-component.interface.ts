export interface IDebtComponent {
  id: number;
  typeCode: number;
  sum: number;
  currencyId: number;
}
