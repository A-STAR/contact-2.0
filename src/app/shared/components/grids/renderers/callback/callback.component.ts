import { Component } from '@angular/core';
import { ICellRendererAngularComp } from 'ag-grid-angular';

import { IRendererCallbackParams } from '../../grids.interface';

@Component({
  selector: 'app-callback-renderer',
  template: `{{ value }}`
})
export class CallbackRendererComponent implements ICellRendererAngularComp {

  private params: IRendererCallbackParams;

  agInit(params: IRendererCallbackParams): void {
    this.params = params;
  }

  refresh(): boolean {
    return false;
  }

  get value(): string {
    return this.params.rendererCallback(this.params.data);
  }

}
