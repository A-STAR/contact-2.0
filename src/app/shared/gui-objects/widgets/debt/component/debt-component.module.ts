import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';

import { DebtComponentGridModule } from './grid/debt-component-grid.module';

import { DebtComponentService } from './debt-component.service';

@NgModule({
  imports: [
    DebtComponentGridModule,
    CommonModule,
  ],
  exports: [
    DebtComponentGridModule,
  ],
  providers: [
    DebtComponentService,
  ]
})
export class DebtComponentModule { }
