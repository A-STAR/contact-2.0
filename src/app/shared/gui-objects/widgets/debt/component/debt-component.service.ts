import { Injectable } from '@angular/core';
import { Observable } from 'rxjs/Observable';

import { IDebtComponent } from './debt-component.interface';

import { DataService } from '../../../../../core/data/data.service';
import { NotificationsService } from '../../../../../core/notifications/notifications.service';

@Injectable()
export class DebtComponentService {
  static MESSAGE_DEBT_COMPONENT_SAVED = 'MESSAGE_DEBT_COMPONENT_SAVED';

  constructor(
    private dataService: DataService,
    private notificationsService: NotificationsService,
  ) {}

  fetchAll(debtId: number): Observable<Array<IDebtComponent>> {
    return this.dataService
      .read('/debts/{debtId}/components', { debtId })
      .map(response => response.components)
      .catch(this.notificationsService.error('errors.default.read').entity('entities.debtComponents.gen.plural').dispatchCallback());
  }

  fetch(debtId: number, debtComponentId: number): Observable<IDebtComponent> {
    return this.dataService
      .read('/debts/{debtId}/components/{debtComponentId}', { debtId, debtComponentId })
      .map(response => response.components[0])
      .catch(this.notificationsService.error('errors.default.read').entity('entities.debtComponents.gen.singular').dispatchCallback());
  }

  create(debtId: number, debtComponent: IDebtComponent): Observable<void> {
    return this.dataService
      .create('/debts/{debtId}/components', { debtId }, debtComponent)
      .catch(this.notificationsService.error('errors.default.create').entity('entities.debtComponents.gen.singular').dispatchCallback());
  }

  update(debtId: number, debtComponentId: number, debtComponent: Partial<IDebtComponent>): Observable<void> {
    return this.dataService
      .update('/debts/{debtId}/components/{debtComponentId}', { debtId, debtComponentId }, debtComponent)
      .catch(this.notificationsService.error('errors.default.update').entity('entities.debtComponents.gen.singular').dispatchCallback());
  }

  delete(debtId: number, debtComponentId: number): Observable<void> {
    return this.dataService
      .delete('/debts/{debtId}/components/{debtComponentId}', { debtId, debtComponentId })
      .catch(this.notificationsService.error('errors.default.delete').entity('entities.debtComponents.gen.singular').dispatchCallback());
  }
}
