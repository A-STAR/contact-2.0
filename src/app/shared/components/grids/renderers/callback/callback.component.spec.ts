import { async as Async, ComponentFixture, TestBed } from '@angular/core/testing';

import { CallbackRendererComponent } from './callback.component';

describe('CallbackCellRendererComponent', () => {
  let fixture: ComponentFixture<CallbackRendererComponent>;
  const mockCallback = (value: any) => (value || { renderCase: 'defaultString' }).renderCase;
  beforeEach(Async(() => {
    TestBed
      .configureTestingModule({
        declarations: [
          CallbackRendererComponent,
        ],
      })
      .compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(CallbackRendererComponent);
  });

  it('should invoke passed renderer callback', () => {
    const agGridParams = {
      rendererCallback: mockCallback,
      value: null,
    };

    const spy = spyOn(agGridParams, 'rendererCallback')
    .and
    .returnValue('');

    fixture.componentInstance.agInit(agGridParams as any);
    fixture.detectChanges();
    expect(fixture.nativeElement).toMatchSnapshot();
    expect(spy).toHaveBeenCalled();
  });

  it('should render transformed value passed through callback', () => {
    fixture.componentInstance.agInit({
      rendererCallback: mockCallback,
      data: {
        renderCase: 'myFancyString',
      },
    } as any);
    fixture.detectChanges();
    expect(fixture.nativeElement).toMatchSnapshot();
  });
});
