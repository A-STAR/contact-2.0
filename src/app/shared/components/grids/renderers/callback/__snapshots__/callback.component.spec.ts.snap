// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CallbackCellRendererComponent should invoke passed renderer callback 1`] = `
<div
  id="root0"
  ng-version="5.2.4"
>
  
</div>
`;

exports[`CallbackCellRendererComponent should render transformed value passed through callback 1`] = `
<div
  id="root1"
  ng-version="5.2.4"
>
  myFancyString
</div>
`;
